receiver/prometheusremotewritereceiver/receiver.go

```diff
@@ -388,23 +389,115 @@ func addNumberDatapoints(datapoints pmetric.NumberDataPointSlice, ls labels.Labe
 		dp.SetDoubleValue(sample.Value)
 
 		attributes := dp.Attributes()
-		for _, l := range ls {
-			if l.Name == "instance" || l.Name == "job" || // Become resource attributes
-				l.Name == labels.MetricName || // Becomes metric name
-				l.Name == "otel_scope_name" || l.Name == "otel_scope_version" { // Becomes scope name and version
-				continue
-			}
-			attributes.PutStr(l.Name, l.Value)
-		}
+		extractAttributes(ls).CopyTo(attributes)
 	}
 }
 
 func addSummaryDatapoints(_ pmetric.SummaryDataPointSlice, _ labels.Labels, _ writev2.TimeSeries) {
 	// TODO: Implement this function
 }
 
-func addHistogramDatapoints(_ pmetric.HistogramDataPointSlice, _ labels.Labels, _ writev2.TimeSeries) {
-	// TODO: Implement this function
+func addExponentialHistogramDatapoints(datapoints pmetric.ExponentialHistogramDataPointSlice, ls labels.Labels, ts writev2.TimeSeries) {
+	for _, histogram := range ts.Histograms {
+		dp := datapoints.AppendEmpty()
+		dp.SetStartTimestamp(pcommon.Timestamp(ts.CreatedTimestamp * int64(time.Millisecond)))
```

Comment:
wdyt about using something like https://github.com/perebaj/opentelemetry-collector-contrib/blob/22f72e63ba071857111d991ebf5f8ccebd540405/receiver/prometheusreceiver/internal/util.go#L71?

Reply:
I tend to agree this don't simplify the work very much, but we have that same conversion spread around 6 different places only in this PR, I'm wondering if this can't increase overtime, ofc this is a nit, I don't have strong opinion on that.